docs/cryptodoc/src/05_06_hss_lms.rst

```diff
@@ -0,0 +1,422 @@
+
+As defined in [RFC8554]_, the public key of an HSS/LMS instance is composed of
+``L`` and the public key of the hypertree's root LMS tree. The
+HSS/LMS secret key format is not defined in [RFC8554]_. Botan defines its own
+secret key format under a private OID. The following describes its byte
+composition in the same syntax as [RFC8554]_:
+
+.. math::
+   \mathtt{SK\_Bytes =\ } &\mathtt{u32str(L)\ ||\ u64str(idx)\ || }
+
+      &\mathtt{u32str(LMSAlgorithmId_{root\_layer})\ ||\ u32str(LMOTSAlgorithmId_{root\_layer})\ || }
+
+      &\mathtt{\dots\ || }
+
+      &\mathtt{u32str(LMSAlgorithmId_{bottom\_layer})\ ||\ u32str(LMOTSAlgorithmId_{bottom\_layer})\ || }
+
+      &\mathtt{SEED_{root\_tree} ||\ I_{root\_tree} }
+
+
+``idx`` is the index of the next signature created using this
+secret key, defining the LMS leaves to use. This entry updates
+after every signature creation. Next, the LMS and LMOTS algorithm IDs are given
+for each level as defined in Tables :ref:`Supported LM-OTS parameter sets
+<pubkey/hss_lms/lm-ots-params>` and :ref:`Supported LMS parameter sets
+<pubkey/hss_lms/lms-params>`. Finally, ``SEED`` and ``I`` of the root LMS tree
+are given. The classes ``HSS_LMS_PublicKeyInternal`` and
+``HSS_LMS_PrivateKeyInternal`` realize the public and secret key, respectively.
+
+Botan's HSS/LMS implementation derives LMS seeds and identifiers
+by the same method Cisco's reference implementation applies. This approach
+is called ``SECRET_METHOD 2`` in the Cisco implementation's configuration.
+``SEED`` and ``I`` of child LMS trees are derived from the values of their
+parents and their position in the hypertree. This operation is similar to the
+pseudorandom key generation method of [RFC8554]_ Appendix A.
+The derivation functions are the following:
+
+.. math::
+   \mathtt{SEED_{child}}\ &\mathtt{= Hash(I_{parent}\ ||\ u32str(q_{parent})\
+   ||\ u16str(0xfffe)\ ||\ u8str(0xff)\ ||\ SEED_{parent})}
+
+   \mathtt{I_{child}}\    &\mathtt{= Hash(I_{parent}\ ||\ u32str(q_{parent})\
+   ||\ u16str(0xffff)\ ||\ u8str(0xff)\ ||\ SEED_{parent})}
+
+:math:`\mathtt{I_{parent}}` and :math:`\mathtt{SEED_{parent}}` are the
+identifier and seed of the parent LMS tree, while :math:`\mathtt{I_{child}}`
+and :math:`\mathtt{SEED_{child}}` are derived for the new child.
+:math:`\mathtt{q_{parent}}` is the parent's LMS leaf index used to sign the
+child LMS tree's public key. Note that since the third entry separates their
+domain, the hash inputs will never collide with
+the ones of the LM-OTS computations of ``x[i]`` and ``C``.
+
+The method ``HSS_LMS_Signature_Operation::sign`` is used for signature creation,
+implementing Algorithm 8 of [RFC8554]_.
+``HSS_LMS_Verification_Operation::is_valid_signature`` provides signature
+verification, as in Section 6.3. of [RFC8554]_.
+
+.. _pubkey/hss_lms/key_gen:
+
+Key Generation
+--------------
+
+HSS/LMS key generation follows Section 6.1. of [RFC8554]_ and is implemented
```

Comment:
I think, if you state here that Botan follows §6.1 in RFC8554 but above in line 263 ff that "Botan's HSS/LMS implementation derives LMS seeds and identifiers by the same method Cisco's reference implementation applies.", then you should comment on the compatibility with the sentence

"In the above algorithm, each LMS public/private key pair generated MUST be generated independently."

of §6.1 in RFC8554. Otherwise people might get confused or ask.